src/main/java/com/dynatrace/openkit/core/communication/BeaconSendingContext.java

```diff
@@ -113,12 +127,12 @@ public void executeCurrentState() {
         nextState = null;
         currentState.execute(this);
 
-        if(nextState != null){ // currentState.execute(...) can trigger state changes
-            // TODO: roland.ettinger log transition
+        if (nextState != currentState) { // currentState.execute(...) can trigger state changes

```

Comment:
nextState != nullptr && nextState != currentState

As far as I know every state sets the next state. If a state would not set nextState it would remain null and in the line currentState = nextState null would be assigned to currentState